health/health_config.c

```diff
@@ -937,6 +984,27 @@ static int health_readfile(const char *filename, void *data) {
                 rc->module_match = string_strdupz(value);
                 rc->module_pattern = simple_pattern_create(rrdcalc_module_match(rc), NULL, SIMPLE_PATTERN_EXACT, true);
             }
+            else if(hash == hash_chart_label && !strcasecmp(key, HEALTH_CHART_LABEL_KEY)) {
+                alert_cfg->chart_labels = string_strdupz(value);
+                if(rc->chart_labels) {
+                    if(strcmp(rrdcalc_chart_labels(rc), value) != 0)
+                        error("Health configuration at line %zu of file '%s' for alarm '%s' has key '%s' twice, once with value '%s' and later with value '%s'.",
+                              line, filename, rrdcalc_name(rc), key, value, value);
+
+                    string_freez(rc->chart_labels);
+                    simple_pattern_free(rc->chart_labels_pattern);
+                }
+
+                {
+                    char *tmp = simple_pattern_trim_around_equal(value);
+                    char *tmp_2 = health_config_add_key_to_values(tmp);
+                    rt->chart_labels = string_strdupz(tmp_2);
```

Comment:
Hello @MrZammler ,

This line has the following issue:

```sh

Thread 32 "HEALTH" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe54df6c0 (LWP 30121)]
0x00000000004a6683 in health_readfile (filename=filename@entry=0x132e3b0 "/etc/netdata/health.d/my_disk.conf", data=data@entry=0xcb9f40) at health/health_config.c:1001
1001                        rt->chart_labels = string_strdupz(tmp_2);
(gdb) bt
#0  0x00000000004a6683 in health_readfile (filename=filename@entry=0x132e3b0 "/etc/netdata/health.d/my_disk.conf", data=data@entry=0xcb9f40) at health/health_config.c:1001
#1  0x0000000000452dc5 in recursive_config_double_dir_load (user_path=user_path@entry=0x139a560 "/etc/netdata/health.d", stock_path=stock_path@entry=0x130fd40 "/usr/lib/netdata/conf.d/health.d", subpath=subpath@entry=0x0, 
    callback=callback@entry=0x4a4663 <health_readfile>, data=data@entry=0xcb9f40, depth=depth@entry=0) at libnetdata/libnetdata.c:1552
#2  0x00000000004a7fa7 in health_readdir (host=host@entry=0xcb9f40, user_path=0x139a560 "/etc/netdata/health.d", stock_path=stock_path@entry=0x130fd40 "/usr/lib/netdata/conf.d/health.d", subpath=subpath@entry=0x0)
    at health/health_config.c:1338
#3  0x00000000004a1074 in initialize_health (host=host@entry=0xcb9f40) at health/health.c:804
#4  0x00000000004a1478 in health_main (ptr=0xb8cbe0) at health/health.c:1022
#5  0x0000000000463694 in netdata_thread_init (ptr=<optimized out>) at libnetdata/threads/threads.c:262
#6  0x00007ffff6badcaa in start_thread () from /lib64/libc.so.6
#7  0x00007ffff6c37d9c in clone3 () from /lib64/libc.so.6
```

I discovered it after to modify a template alert:

```sh
template: disk_example
      on: disk.busy
  lookup: sum -1m unaligned absolute of busy
   units: seconds
chart labels: device=sda
#  charts: !*sdc* *
   every: 1m
    warn: $this > 4000
    info: testing with disks
      to: sysadmin
```

to a simple alarm:

```sh
alarm: disk_example
      on: disk_busy.sda
  lookup: sum -1m unaligned absolute of busy
   units: seconds
chart labels: device=sda
#  charts: !*sdc* *
   every: 1m
    warn: $this > 4000
    info: testing with disks
      to: sysadmin
```

Please, take a look when you have a time.

Reply:
Thanks @thiagoftsm ! Looking!

Reply:
"I will not copy paste again", "I will not copy paste again" .... Thanks Thiago!